src/develop/masks/detail.c

```diff
@@ -16,6 +16,74 @@
+  A threshold value of 0.0 means bypassing.
+  
+  So the first important point is:
+  We make sure taking the input data for the DM right from the demosaicer for normal raws
+  or from rawprepare in case of monochromes. This means some additional housekeeping for the
+  pixelpipe.
+  If any mask in any module selects a threshold of != 0.0 we leave a flag in the pipe struct
+  telling a) we want a DM and b) we want it from either demosaic or from rawprepare.
+  If such a flag has not been previously set we will force a pipeline reprocessing.
+  
+  gboolean dt_dev_write_rawdetail_mask(dt_dev_pixelpipe_iop_t *piece, float *const rgb, const dt_iop_roi_t *const roi_in, const int mode, const float wb[3]);
+  or it's _cl equivalent write a preliminary mask holding signal-change values for every pixel.
+  These mask values are calculated as
+  a) get Y0 for every pixel
+  b) apply a scharr operator on it
+
+  This raw detail mask (RM) is not scaled but only cropped to the roi of the writing module (demosaic
+  or rawprepare).
+  The pipe gets roi copy of the writing module so we can later scale/distort the LM.
+
+  Calculating the RM is done for performance and lower mem pressure reasons, so we don't have to
+  pass full data to the module. Also the RM can be used by other modules. 
+ 
+  If a mask uses the details refinement step it takes the raw details mask RM and calculates an
+  intermediate mask (IM) which is still not scaled but has the roi of the writing module.
+ 
+  For every pixel we calculate the IM value via a sigmoid function with the threshold and RM as parameters.
+
+  At last the IM is slightly blurred to avoid hard transitions, as there still is no scaling we can use
+  a constant sigma. As the blur_9x9 is pretty fast both in openmp/cl code paths - much faster than dt
+  gaussians - it is used here.
+  Now we have an unscaled detail mask which requires to be transformed through the pipeline using
+
+  float *dt_dev_distort_detail_mask(const dt_dev_pixelpipe_t *pipe, float *src, const dt_iop_module_t *target_module)
+
+  returning a pointer to a distorted mask (DT) with same size as used in the module wanting the refinement.
+  This DM is finally used to refine the original mask.
+
+  All other refinements and parametric parameters are untouched.
+
+  Some additional comments:
+  1. intentionally this details mask refinement has only been implemented for raws. Especially for compressed
+     inmages like jpegs or 8bit input the algo didn't work as good because of input precision and compression artefacts.
+  2. In the gui the slider is above the rest of the refinemt sliders to emphasize that blurring & feathering use the
+     mask corrected by detail refinemnt.
+  3. Of course credit goes to Ingo @heckflosse from rt team for the original idea. (in the rt world this is knowb
+     as details mask)
+  4. Thanks to rawfiner for pointing out how to use Y0 and scharr for better maths.
```

Comment:
no need to thank me in the code like that :-)

Reply:
Just a reminder :-)